docs/reference/mapping/types/dense-vector.asciidoc

```diff
@@ -111,6 +111,39 @@ PUT my-index-2
 efficient kNN search. Like most kNN algorithms, HNSW is an approximate method
 that sacrifices result accuracy for improved speed.
 
+[[dense-vector-quantization]]
+==== Automatically quantize vectors for kNN search
+
+The `dense_vector` type supports quantization to reduce the memory footprint required when searching `float` vectors.
+Currently the only quantization method supported is `byte` and provided vectors `element_type` must be `float`.
```

Comment:
I think we want to add a line here that this is only supported for HNSW right?

Reply:
Maybe? But the fact that its under the `index_options` already implies this.